avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroTypeDeserializer.java

```diff
@@ -57,6 +54,13 @@ public Object deserializeTypedFromScalar(JsonParser p, DeserializationContext ct
 
     @Override
     public Object deserializeTypedFromAny(JsonParser p, DeserializationContext ctxt) throws IOException {
+        AnnotationIntrospector ai = ctxt.getAnnotationIntrospector();
+        if (ai != null && _property != null) {
+            JsonDeserializer<?> deser = ctxt.deserializerInstance(_property.getMember(), ai.findDeserializer(_property.getMember()));
```

Comment:
Yeah, this was more in the "it _worked_" category rather than the "this is how to fix it" category.

I think I found a better solution, about to push up some changes.